OpenAPI/kiota/tutorials/dotnet-dependencyinjection.md

```diff
@@ -0,0 +1,288 @@
+  /repos/{owner}/{repo}/releases/latest:
+    get:
+      summary: Get the latest release
+      parameters:
+        - name: owner
+          in: path
+          required: true
+          description: The owner of the repository
+          schema:
+            type: string
+        - name: repo
+          in: path
+          required: true
+          description: The repository name
+          schema:
+            type: string
+      responses:
+        200:
+          description: Success!
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/github.release"
+        404:
+          description: Not found
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/github.error"
+
+components:
+  schemas:
+    github.release:
+      type: object
+      properties:
+        id:
+          type: integer
+          description: The release ID
+        tag_name:
+          type: string
+          description: The release tag name
+        name:
+          type: string
+          description: The release name
+        body:
+          type: string
+          description: The release body
+        created_at:
+          type: string
+          format: date-time
+          description: The release creation date
+        published_at:
+          type: string
+          format: date-time
+          description: The release publication date
+    github.error:
+      type: object
+      properties:
+        message:
+          type: string
+          description: The error message
+        documentation_url:
+          type: string
+          description: The error documentation URL
+```
+
+You can then use the [Kiota](/openapi/kiota/install?tabs=bash&wt.mc_id=SEC-MVP-5004985#install-as-net-tool) command line tool to generate the API client classes. We are using the [deserializer](/openapi/kiota/using#--deserializer---ds) and [serializer](/openapi/kiota/using#--serializer---s) options to specify the serialization format, to only use JSON serialization.
+
+```bash
+kiota generate -l csharp -d github-releases.yml -c GitHubClient -n GitHub.ApiClient -o ./GitHub --deserializer Microsoft.Kiota.Serialization.Json.JsonParseNodeFactory --serializer Microsoft.Kiota.Serialization.Json.JsonSerializationWriterFactory
+```
+
+## Create extension methods
+
+Create a new file named _KiotaServiceCollectionExtensions.cs_ in the root of your project and add the following code.
+
+```csharp
+using Microsoft.Kiota.Http.HttpClientLibrary.Middleware;
+using Microsoft.Kiota.Http.HttpClientLibrary.Middleware.Options;
+
+/// <summary>
+/// Service collection extensions for Kiota handlers.
+/// </summary>
+public static class KiotaServiceCollectionExtensions
+{
+    /// <summary>
+    /// Adds the Kiota handlers to the service collection.
+    /// </summary>
+    /// <param name="services"><see cref="IServiceCollection"/> to add the services to</param>
```

Comment:
add cref to this file https://github.com/MicrosoftDocs/openapi-docs/blob/main/cspell.json